### PR TITLE
 lint: fix if and loops are not inside braces and  remove redundant calls to smart pointer’s .get() method

### PR DESCRIPTION
**Pull request by zyfjeff**
1. remove redundant calls to smart pointer’s .get() method.
 2. add braces for bodies of if statements and loops (for, do while, and while) are not inside braces

*Description*: fix if and loops are not inside braces and  remove redundant calls to smart pointer’s .get() method.
*Risk Level*: lower
*Testing*: unittests
*Docs Changes*: N/A
*Release Notes*: n/A
[Optional Fixes #Issue] #4863

### CODE REVIEW

File: source/extensions/filters/network/mongo_proxy/codec_impl.cc [hunk 1]
```diff
@@ -426,8 +426,9 @@ bool DecoderImpl::decode(Buffer::Instance& data) {
 }
 
 void DecoderImpl::onData(Buffer::Instance& data) {
-  while (data.length() > 0 && decode(data))
+  while (data.length() > 0 && decode(data)) {
     ;
```
Comment (mattklein123) [inline source/extensions/filters/network/mongo_proxy/codec_impl.cc]: You can remove this now.

**Comment by lizan** [comment]
tidy is catching error involved in diff fixed by #4933, @zyfjeff can you merge master?

**Comment by zyfjeff** [comment]
@lizan  merged

**Comment by mattklein123** [comment]
@zyfjeff LGTM but needs a master merge.

**Comment by zyfjeff** [comment]
@mattklein123  merged

**Comment by zyfjeff** [comment]
can you PTAL when you get time?  @mattklein123

**Pull request merged**